### PR TITLE
Added CI support for redis ~> 5.0 gem.

### PR DESCRIPTION
**Pull request by Jack12816**
**- What is it good for**

As a sub-level dependency, of eg. [sidekiq-unique-jobs](https://rubygems.org/gems/sidekiq-unique-jobs), this gem blocks the upgrade to a newer Sidekiq version (>= 7.0). As the gemspec of brpoplpush-redis_script tells it is compatible with the redis gem lower OR equal to version 5.0. This is a bit "unhappily" expressed with `"<= 5.0"`. I guess it was intended to mark an incompatibility with 5.0, so `< 5` would be appropriate. See: https://github.com/brpoplpush/brpoplpush-redis_script/issues/23

**- What I did**

Looks like this gem is compatible to the redis gem in version 5.0. So I added an Appraisals entry for it and CI (enabled on my fork, I dunno why it won't start here at this PR), [and it looks like it works without issues](https://github.com/Jack12816/brpoplpush-redis_script/actions/runs/3190735888/jobs/5206232745).

**- A picture of a cute animal (not mandatory but encouraged)**

images

---

Side note: When this change is fine for you @brpoplpush, could you release it as version `0.1.3`? This would be helpful without causing other glitches. :)

### CODE REVIEW

**Comment by Jack12816** [comment]
Ping @mhenrixon @mberlanda? :)

**Review by mhenrixon — APPROVED** [review]

File: brpoplpush-redis_script.gemspec [hunk 1]
```diff
@@ -31,7 +31,7 @@ Gem::Specification.new do |spec|
   spec.required_ruby_version = ">= 2.5.0"
 
   spec.add_dependency "concurrent-ruby", "~> 1.0", ">= 1.0.5"
-  spec.add_dependency "redis", ">= 1.0", "<= 5.0"
+  spec.add_dependency "redis", ">= 1.0", "< 6"
```
Comment (mhenrixon) [inline brpoplpush-redis_script.gemspec:34]: This should possibly be the `redis-client` due to how sidekiq works under the hood in v7. But like I commented in the other PR. This makes me want to move the code back to sidekiq-unique-jobs. 

I'll have to think about this a bit...

Comment (clemens) [reply brpoplpush-redis_script.gemspec:34]: Could we maybe at least release a fixed version where the constraint is (more correctly) `< 5`? That would at least fix the main issue.

**Comment by mhenrixon** [comment]
This should have been fixed in the previous PR. At least the code still works with sidekiq 7. 

Please let me know if that's not the case and I will fix it.

**Comment by Jack12816** [comment]
Nice. Sadly, the redis-5 appraisal config is missing upstream. :(

```ruby
appraise 'redis-5.x' do
  gem 'redis', '~> 5.0'
end
```

**Comment by mhenrixon** [comment]
Ah, I knew there was something I was missing! I'll get it added.

**Comment by mhenrixon** [comment]
#27

**Pull request closed without merging**